pkg/labels/fake_applicator.go

```diff
@@ -102,21 +105,73 @@ func (app *fakeApplicator) GetMatches(selector labels.Selector, labelType Type)
 	return results, nil
 }
 
-func (app *fakeApplicator) WatchMatches(selector labels.Selector, labelType Type, quitCh chan struct{}) chan []Labeled {
+func (app *fakeApplicator) WatchMatches(selector labels.Selector, labelType Type, quitCh <-chan struct{}) chan []Labeled {
 	ch := make(chan []Labeled)
+	timer := time.NewTimer(time.Duration(0))
 	go func() {
 		for {
+			select {
+			case <-quitCh:
+				return
+			case <-timer.C:
+			}
+			// upper bound on request rate
+			timer.Reset(100 * time.Millisecond)
+
 			res, _ := app.GetMatches(selector, labelType)
 			select {
 			case <-quitCh:
 				return
 			case ch <- res:
 			}
+
 		}
 	}()
 	return ch
 }
 
+func (app *fakeApplicator) WatchMatchDiff(
+	selector labels.Selector,
+	labelType Type,
+	quitCh <-chan struct{},
+) <-chan *LabeledChanges {
+	outCh := make(chan *LabeledChanges)
+
+	go func() {
+		defer close(outCh)
+
+		logger := logging.DefaultLogger
+		inCh := app.WatchMatches(selector, labelType, quitCh)
+		diffCh := watchDiffLabels(inCh, quitCh, logger)
+		for {
```

Comment:
What does this loop do? As far as I can tell, it copies from one channel to another. Can you just return `diffCh`?